### PR TITLE
Make show series return IDs and work with limit and offset.

### PR DESCRIPTION
**Pull request by pauldix**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -2048,14 +2044,49 @@ func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement,
...
 			}
 		}
+		// make the id the first column
+		r.Columns = append([]string{"id"}, r.Columns...)
 
 		// Append the row to the result.
 		result.Series = append(result.Series, r)
 	}
 
+	if stmt.Limit > 0 {
```
Comment (dgnorton) [inline server.go]: should this be `if stmt.Limt > 0 || stmt.Offset > 0 {`?

Comment (pauldix) [inline server.go]: ah, yes

File: server.go [hunk 2]
```diff
@@ -2048,14 +2044,49 @@ func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement,
...
+		result.Series = s.filterShowSeriesResult(stmt.Limit, stmt.Offset, result.Series)
+	}
+
 	return result
 }
 
+// filterShowSeriesResult will limit the number of series returned based on the limit and the offset.
+// Unlike limit and offset on SELECT statements, the limit and offset don't apply to the number of Rows, but
+// to the number of total Values returned, since each Value represents a unique series.
+func (s *Server) filterShowSeriesResult(limit, offset int, rows influxql.Rows) influxql.Rows {
```
Comment (dgnorton) [inline server.go]: Doesn't need to be a method of `Server`.  Or, if left as a method, don't shadow the receiver `s` below.

**Comment by dgnorton** [comment]
Couple inline comments.  Ohterwise, lgtm :shipit"

**Pull request merged**